stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -565,7 +593,11 @@ swift::swift_conformsToProtocol(const Metadata *type,
   unsigned failedGeneration = ConformanceCacheGeneration;
 
   // If we didn't have an up-to-date cache entry, scan the conformance records.
+#if defined(_MSC_VER)
+  C.SectionsToScanLock.lock();
+#else
   pthread_mutex_lock(&C.SectionsToScanLock);
```

Comment:
Note also the work-in-progress in PR https://github.com/shawnce/swift/pull/7. I only need to finish off some header docs and likely some spit and polish before opening a PR back into Apple's repo. I had hope #1731 would have been in by now.